packages/graphql/src/translate/create-projection-and-params.ts

```diff
@@ -251,12 +251,15 @@ function createProjectionAndParams({
         }
 
         if (relationField) {
+            // WARN: In some test this node is undefined. If this is intended, we should update typings to reflect that
             const referenceNode = context.neoSchema.nodes.find((x) => x.name === relationField.typeMeta.name) as Node;
             const nodeMatchStr = `(${chainStr || varName})`;
             const inStr = relationField.direction === "IN" ? "<-" : "-";
             const relTypeStr = `[:${relationField.type}]`;
             const outStr = relationField.direction === "OUT" ? "->" : "-";
-            const nodeOutStr = `(${param}:${referenceNode?.name})`;
+            const labels =
+                referenceNode?.nodeDirective?.getLabelsString(referenceNode.name) || `:${referenceNode?.name}`;
```

Comment:
This `||` is scattered throughout this change and I fear that it could be brittle and or missed in particular use cases, what do you think about creating a function that does both the `.getLabelsString(referenceNode.name)` and the `|| :${referenceNode?.name}`; ?

Reply:
We were discussing the possibility of this being part of the Node class (something like `referenceNode.getLabels()`)
thoughts?

Reply:
I think that would be better as everything will be in one place, that being said does it need to be a function? Could it be defined once in the `constructor` of each node?

Reply:
Function, a get function or setting in a constructor, I would say that any of those should be fine if we decide to go for this approach of adding it to Node class